modules/compatibility/src/test/java/org/apache/ignite/compatibility/testframework/junits/CompatibilityTestIgniteNodeRunner.java

```diff
@@ -0,0 +1,172 @@
+
+        X.println("Starting Ignite Node... Args=" + Arrays.toString(args));
+
+        if (args.length < 3) {
+            throw new IllegalArgumentException("At least four arguments expected:" +
+                " [path/to/closure/file] [ignite-instance-name] [node-uuid] [sync-node-uuid] [optional/path/to/closure/file]");
+        }
+
+        IgniteConfiguration cfg = CompatibilityTestsFacade.getConfiguration();
+
+        IgniteInClosure<IgniteConfiguration> cfgClos = readClosureFromFileAndDelete(args[0]);
+
+        cfgClos.apply(cfg);
+
+        final UUID id = UUID.fromString(args[2]);
+        final UUID syncId = UUID.fromString(args[3]);
+
+        // Ignite instance name and id must be set according to arguments
+        // it's used for nodes managing: start, stop etc.
+        cfg.setIgniteInstanceName(args[1]);
+        cfg.setNodeId(id);
+
+        final Ignite ignite = Ignition.start(cfg);
+
+        // If 'id' equals 'syncId' then the starting node is the first node and
+        // there was no need to check the join to topology.
+        if (!id.equals(syncId)) {
+            GridTestUtils.waitForCondition(new GridAbsPredicate() {
+                @Override public boolean apply() {
+                    boolean found = ignite.cluster().node(syncId) != null;
+
+                    if (found)
+                        X.println(IgniteCompatibilityAbstractTest.SYNCHRONIZATION_LOG_MESSAGE_JOINED + id);
+
+                    return found;
+                }
+            }, 20);
+        }
+
+        // It needs to set private static field 'ignite' of the IgniteNodeRunner class via reflection
+        GridTestUtils.setFieldValue(new IgniteNodeRunner(), "ignite", ignite);
+
+        if (args.length == 5) {
+            IgniteInClosure<Ignite> iClos = readClosureFromFileAndDelete(args[4]);
+
+            iClos.apply(ignite);
+        }
+
+        X.println(IgniteCompatibilityAbstractTest.SYNCHRONIZATION_LOG_MESSAGE_PREPARED + id);
+    }
+
+    /**
+     * Stores {@link IgniteInClosure} to file as xml.
+     *
+     * @param clos IgniteInClosure.
+     * @return A name of file where the closure was stored.
+     * @throws IOException In case of an error.
+     * @see #readClosureFromFileAndDelete(String)
+     */
+    @Nullable public static String storeToFile(@Nullable IgniteInClosure clos) throws IOException {
+        if (clos == null)
+            return null;
+
+        String fileName = IGNITE_CLOSURE_FILE + clos.hashCode();
+
+        storeToFile(clos, fileName);
+
+        return fileName;
+    }
+
+    /**
+     * Stores {@link IgniteInClosure} to file as xml.
+     *
+     * @param clos IgniteInClosure.
+     * @param fileName A name of file where the closure was stored.
+     * @throws IOException In case of an error.
+     * @see #readClosureFromFileAndDelete(String)
+     */
+    public static void storeToFile(@NotNull IgniteInClosure clos, @NotNull String fileName) throws IOException {
```

Comment:
Why doesn't IgniteInClosure have generic parameter?

Reply:
Do you think this is required?
I think no because given instance will be serialized to a file, without any type information.

Reply:
Is `storeToFile` common method? Or it will be used only in compatibility module with specific closure types?

If first - we should use  
`public static void <T> storeToFile(@NotNull IgniteInClosure<T> clos, @NotNull String fileName) throws IOException {`
If second:
`public static void storeToFile(@NotNull IgniteInClosure<Ignite> clos, @NotNull String fileName) throws IOException {`

Reply:
It will be used only in the compatibility module with specific closure types:
IgniteInClosure<IgniteConfiguration>
IgniteInClosure<Ignite>